homeassistant/components/geocaching/__init__.py

```diff
@@ -36,15 +42,21 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     if DOMAIN not in config:
         return True
 
-    GeocachingFlowHandler.async_register_implementation(
+    await async_import_client_credential(
         hass,
-        GeocachingOAuth2Implementation(
-            hass,
-            client_id=config[DOMAIN][CONF_CLIENT_ID],
-            client_secret=config[DOMAIN][CONF_CLIENT_SECRET],
-            name="Geocaching",
+        DOMAIN,
+        ClientCredential(
+            config[DOMAIN][CONF_CLIENT_ID],
+            config[DOMAIN][CONF_CLIENT_SECRET],
         ),
     )
+    _LOGGER.warning(
```

Comment:
We don't need to deprecate this integration, as it has never been stable. We can simply remove the whole configuration schema.

Reply:
Great! geocaching folks this will break your dev instances.

Reply:
I won't, as getting a token is nearly impossible (hence this integration supports account linking).

Reply:
Awesome, even better, that is helpful context.  This has all been removed.

While we're discussing this I have two other considerations so if you have any other thoughts on these let me know:
- I was thinking about supporting integration discovery based on application credentials, but have not added it yet  I noticed this integration does discovery for itself based on configuration.yaml but that has been removed now.
- May be possible to further simplify test setup, but just leaving it as is for now.  This is based on what I have proposed for the scaffold script to generate.

Reply:
> I was thinking about supporting integration discovery based on application credentials, but have not added it yet I noticed this integration does discovery for itself based on configuration.yaml but that has been removed now.

Yes, it did that because it  was next logical step. One adds app credentials to the config, restarts HA to make it effective, next adds an integration entry. The "discovery"  was mainly helpful to aid/shorten the flow to the last step.

For this integration, it doesn't matter that much. Removing it, is fine.

I guess in the future it would become less of an issue (when the application credentials can be "part of" the original flow).

Reply:
Good point, users won't really start in app creds panel.

Reply:
I will test this in the morning (mine that is); I can't right now